src/contexts/UserInfoContext.tsx

```diff
@@ -24,28 +27,25 @@ export const useUserInfo = () => {
 };
 
 export const UserInfoProvider = ({ children }: PropsWithChildren) => {
-  const [userInfo, setUserInfo] = useState<UserInfo>(() => ({
-    name: sessionStorage.getItem('name') ?? '',
-    email: sessionStorage.getItem('email') ?? '',
-    token: sessionStorage.getItem('token') ?? '',
-  }));
+  const [userInfo, setUserInfo] = useState<UserInfoData>(() => {
+    const parsed = parseSessionStorage();
+    return parsed ? parsed : EMPTY_USER_INFO;
+  });
```

Comment:
useState 내에서 sessionStorage를 가져오는 것보다는 따로 선언해주는게 좋겟어용!
```suggestion
    const parsed = parseSessionStorage();
    const [userInfo, setUserInfo] = useState<UserInfoData>(parsed ? parsed : EMPTY_USER_INFO);
```